pyanaconda/format_dasd.py

```diff
@@ -0,0 +1,195 @@
+        """Is DASD formatting supported on this machine?"""
+        return arch.is_s390()
+
+    @property
+    def report(self):
+        """Signal for the progress reporting.
+
+        Emits messages during the formatting.
+        """
+        return self._report
+
+    @property
+    def dasds(self):
+        """List of found DASDs to format."""
+        return self._dasds
+
+    @property
+    def dasds_summary(self):
+        """Returns a string summary of DASDs to format."""
+        return "\n".join(map(self.get_dasd_info, self.dasds))
+
+    def get_dasd_info(self, disk):
+        """Returns a string with description of a DASD,"""
+        return "/dev/" + disk.name + " (" + disk.busid + ")"
+
+    def _is_dasd(self, disk):
+        """Is it a DASD disk?"""
+        return disk.type == "dasd"
+
+    def _is_unformatted_dasd(self, disk):
+        """Is it an unformatted DASD?"""
+        return self._is_dasd(disk) and blockdev.s390.dasd_needs_format(disk.busid)
+
+    def _is_ldl_dasd(self, disk):
+        """Is it an LDL DASD?"""
+        return self._is_dasd(disk) and blockdev.s390.dasd_is_ldl(disk.name)
+
+    def _get_unformatted_dasds(self, disks):
+        """Returns a list of unformatted DASDs."""
+        result = []
+
+        if not self._can_format_unformatted:
+            log.debug("We are not allowed to format unformatted DASDs.")
+            return result
+
+        for disk in disks:
+            if self._is_unformatted_dasd(disk):
+                log.debug("Found unformatted DASD: %s", self.get_dasd_info(disk))
+                result.append(disk)
+
+        return result
+
+    def _get_ldl_dasds(self, disks):
+        """Returns a list of LDL DASDs."""
+        result = []
+
+        if not self._can_format_ldl:
+            log.debug("We are not allowed to format LDL DASDs.")
+            return result
+
+        for disk in disks:
+            if self._is_ldl_dasd(disk):
+                log.debug("Found LDL DASD: %s", self.get_dasd_info(disk))
+                result.append(disk)
+
+        return result
+
+    def update_restrictions(self, data):
+        """Read kickstart data to update the restrictions."""
+        self._can_format_unformatted = data.zerombr.zerombr
+        self._can_format_ldl = data.clearpart.cdl
+
+    def search_disks(self, disks):
+        """Search for a list of disks for DASDs to format."""
+        self._dasds = list(set(self._get_unformatted_dasds(disks) + self._get_ldl_dasds(disks)))
+
+    def should_run(self):
+        """Should we run the formatting?"""
+        return bool(self._dasds)
+
+    def do_format(self, disk):
+        """Format a disk."""
+        try:
+            self.report.emit(_("Formatting %s") % self.get_dasd_info(disk))
+            blockdev.s390.dasd_format(disk.name)
+        except blockdev.S390Error as err:
+            self.report.emit(_("Failed formatting %s") % self.get_dasd_info(disk))
+            log.error(err)
+
+    def run(self, storage, data):
```

Comment:
I would append to docstring that his could be run in separate thread. It could be valuable information for the future.

Reply:
Fixed.